pkg/syncer/spec/mutators/deployment.go

```diff
@@ -17,18 +17,30 @@ limitations under the License.
 package mutators
 
 import (
+	"crypto/sha256"
+	"fmt"
 	"net/url"
 
+	"github.com/kcp-dev/logicalcluster"
+
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/tools/cache"
 	utilspointer "k8s.io/utils/pointer"
 )
 
+const (
+	secretNamePrefix     = "kcp-"
+	ByWorkspaceIndexName = "syncer-spec-byWorkspace-index" // will go away with scoping
+)
+
 type DeploymentMutator struct {
-	upstreamURL *url.URL
+	upstreamURL  *url.URL
+	secretLister cache.Indexer
```

Comment:
this is not a lister

Reply:
please also put upstream or downstream into the name

Reply:
that code is gone now, done